cyme/core/simd_vector/detail/powerpc64/simd_wrapper_vmx.ipp

```diff
@@ -0,0 +1,1789 @@
+      for 2^k in double, the computation is done using a C trick.
+       specialisation double,cyme::vmx,2 regs
+     */
+    template<>
+    forceinline simd_trait<double,cyme::vmx,2>::register_type
+    _mm_twok<double,cyme::vmx,2>(simd_trait<int,cyme::vmx,2>::register_type xmm0){
+	vector int tmp = vec_splats(1023);
+	xmm0.r0 = vec_add(xmm0.r0,tmp);
+	xmm0.r1 = vec_add(xmm0.r1,tmp);
+	xmm0.r0 = vec_sl(xmm0.r0,vec_splats((unsigned int)20));
+	xmm0.r1 = vec_sl(xmm0.r1,vec_splats((unsigned int)20));
+	return simd_trait<double,cyme::vmx,2>::register_type((vector double)xmm0.r0,
+							     (vector double)xmm0.r1);
+    }
+
+    /**
+      Returns a vector containing 2^k where k is a vector of integer, because there is no builtin function
+      for 2^k in double, the computation is done using a C trick.
+       specialisation double,cyme::vmx,4 regs
+     */
+    template<>
+    forceinline simd_trait<double,cyme::vmx,4>::register_type
+    _mm_twok<double,cyme::vmx,4>(simd_trait<int,cyme::vmx,4>::register_type xmm0){
+	vector int tmp = vec_splats(1023);
+	xmm0.r0 = vec_add(xmm0.r0,tmp);
+	xmm0.r1 = vec_add(xmm0.r1,tmp);
+	xmm0.r2 = vec_add(xmm0.r2,tmp);
+	xmm0.r3 = vec_add(xmm0.r3,tmp);
+	xmm0.r0 = vec_sl(xmm0.r0,vec_splats((unsigned int)20));
+	xmm0.r1 = vec_sl(xmm0.r1,vec_splats((unsigned int)20));
+	xmm0.r2 = vec_sl(xmm0.r2,vec_splats((unsigned int)20));
+	xmm0.r3 = vec_sl(xmm0.r3,vec_splats((unsigned int)20));
+	return simd_trait<double,cyme::vmx,4>::register_type((vector double)xmm0.r0,
+							     (vector double)xmm0.r1,
+							     (vector double)xmm0.r2,
+							     (vector double)xmm0.r3);
+    }
+
+   /**
+      Extract the exponent of floating-point exponent (64-bit) elements and store the results in dst.
+      Because we are casting from vector2double -> vector4int, must use specific mask and tmp_sub vectors.
+       specialisation double,cyme::vmx,1 regs
+     */
+    template<>
+    forceinline simd_trait<double,cyme::vmx,1>::register_type
+    _mm_ge<double,cyme::vmx,1>(simd_trait<double,cyme::vmx,1>::register_type xmm0){
+	vector signed int tmp = (vector signed int)xmm0;
+	vector signed int mask = {(signed int)0xffffffff,(signed int)0x00000000,
+				  (signed int)0xffffffff,(signed int)0x00000000};
+	vector signed int tmp_sub = {1023,0,1023,0};
+	tmp = vec_and(tmp,mask);
+	tmp = vec_sr(tmp,vec_splats((unsigned int)20));
+	tmp = vec_sub(tmp,tmp_sub);
+	return vec_ctd(tmp);
+    }
+
+   /**
+      Extract the exponent of floating-point exponent (64-bit) elements and store the results in dst.
+      Because we are casting from vector2double -> vector4int, must use specific mask and tmp_sub vectors.
+       specialisation double,cyme::vmx,2 regs
+     */
+    template<>
+    forceinline simd_trait<double,cyme::vmx,2>::register_type
+    _mm_ge<double,cyme::vmx,2>(simd_trait<double,cyme::vmx,2>::register_type xmm0){
+	vector signed int tmp0 = (vector signed int)xmm0.r0;
```

Comment:
I tested your work on power8 there is an error because the power8 is big endian where as the power7 is little endian.
So we must use the following macro __LITTLE_ENDIAN__ and  __BIG_ENDIAN__

So please modify _mm_ge unroll 1,2 and 4 and __mm_gf unroll1 2 4 for double only. You just permute the mask and the constant as I ma indicating in the next example 
 

    #ifdef __LITTLE_ENDIAN__ // POWER8, just permute the masks  
            vector signed int mask = {(signed int)0x00000000,(signed int)0xffffffff,
                                                     (signed int)0x00000000,(signed int)0xffffffff};
            vector signed int tmp_sub = {0,1023,0,1023};
    #endif

    #ifdef __BIG_ENDIAN__ // POWER7, the current version of your work
       vector signed int mask = {(signed int)0xffffffff,(signed int)0x00000000,
                                                (signed int)0xffffffff,(signed int)0x00000000};
       vector signed int tmp_sub = {1023,0,1023,0};
    #endif